src/System.IO.FileSystem/tests/FileSystemTest.cs

```diff
@@ -21,12 +21,27 @@ protected static bool CanCreateSymbolicLinks
         {
             get
             {
-                var path = Path.GetTempFileName();
-                var linkPath = path + ".link";
-                var ret = MountHelper.CreateSymbolicLink(linkPath, path, isDirectory: false);
+                // Verify file symlink creation
+                string path = Path.GetTempFileName();
+                string linkPath = path + ".link";
+                if (!MountHelper.CreateSymbolicLink(linkPath, path, isDirectory: false) || !File.Exists(linkPath))
+                {
+                    return false;
```

Comment:
If a symlink can't be created, we still want to delete the temporary file, right?

And if it actually is possible that CreateSymbolicLink could return true and then File.Exists could return false, we'd also need to delete the created symlink.